### PR TITLE
Custom gSPSegment and improve sequence numbering

### PR DESCRIPTION
**Pull request by sauraen**
- Add 1 to sequence number in export so that in-game sequence number is as expected
- Add two optional custom gSPSegment addresses to material editor

### CODE REVIEW

File: fast64_internal/oot/oot_f3d_writer.py [hunk 1]
```diff
@@ -601,6 +605,11 @@ def drawOOTMaterialDrawLayerProperty(layout, matDrawLayerProp, suffix):
 			i = 8 + colIndex * 3 + rowIndex
 			name = "Segment " + format(i, 'X') + " " + suffix
 			col.prop(matDrawLayerProp, "segment" + format(i, 'X'), text = name)
+		name = "Custom " + str(colIndex) + " " + suffix
```
Comment (Dragorn421) [inline fast64_internal/oot/oot_f3d_writer.py:608]:
```suggestion
		name = "Jump to (" + str(colIndex + 1) + ") " + suffix
```
I think "custom" is a pretty bad name to use for everything (way too generic), maybe change `custom` to `customJump`

File: fast64_internal/oot/oot_f3d_writer.py [hunk 2]
```diff
@@ -630,6 +639,10 @@ class OOTDynamicMaterialDrawLayerProperty(bpy.types.PropertyGroup):
 	segmentB : bpy.props.BoolProperty()
 	segmentC : bpy.props.BoolProperty()
 	segmentD : bpy.props.BoolProperty()
+	custom0 : bpy.props.BoolProperty()
+	custom0_seg : bpy.props.StringProperty()
```
Comment (Dragorn421) [inline fast64_internal/oot/oot_f3d_writer.py:643]:
```suggestion
	custom0_seg : bpy.props.StringProperty(description="Segment address of a display list to jump to")
```

**Review by Dragorn421 — CHANGES_REQUESTED** [review]
Sorry, didn't see the PRs until right now by accident. Didn't enable notifications for the repo

Ideally we would want to allow an arbitrary amount of such "custom" jumps but two is probably plenty enough, and a CollectionProperty needs more work

**Comment by Dragorn421** [comment]
hijacking this pr to test the webhook

**Comment by Zelllll** [comment]
Yep I agreed with Dragorn's suggestions, although eventually I'd like to simply remove all the check boxes too and just let users choose as many segments as they want instead, I think it would be worth breaking compatibility with existing stuff to make that change.

**Comment by Dragorn421** [comment]
First

**Comment by sauraen** [comment]
I started working on changing over the segments to a collection, but I realized that the fast64 infrastructure for creating GUI for editing collections is based on the collection in question being some sub-property of an object. This collection would be a sub-property of a material, which means I'd have to make completely new GUI for this or make substantial changes to the existing infrastructure. Combined with the fact that this will break scenes (or the extra work of checking for both types of segment commands), I don't think it's worth it.

I've changed the names to "custom call". It's more like a function call than a jump, as execution returns to the current DL.

**Comment by Zelllll** [comment]
Hm I was thinking using the same gui as the object list, just simply have an add button for a segment, and also allowing you to reorder them so they get called in the specific order you want

**Comment by sauraen** [comment]
Right, so I tried that, and it does not work, because the GUI infrastructure for displaying the elements of a collection with the add/remove/up/down buttons etc. all depends on the collection you're editing being a sub-property of an object in the scene. This collection would be a sub-property of a material, not an object. So it is not compatible with that GUI infrastructure.

File: fast64_internal/oot/oot_f3d_writer.py [hunk 3]
```diff
@@ -630,6 +639,10 @@ class OOTDynamicMaterialDrawLayerProperty(bpy.types.PropertyGroup):
 	segmentB : bpy.props.BoolProperty()
 	segmentC : bpy.props.BoolProperty()
 	segmentD : bpy.props.BoolProperty()
+	customCall0 : bpy.props.BoolProperty()
+	customCall0_seg : bpy.props.StringProperty()
+	customCall1 : bpy.props.BoolProperty()
+	customCall1_seg : bpy.props.StringProperty()
```
Comment (Dragorn421) [inline fast64_internal/oot/oot_f3d_writer.py:645]:
```suggestion
	customCall0_seg : bpy.props.StringProperty(description="Segment address of a display list to call")
	customCall1 : bpy.props.BoolProperty()
	customCall1_seg : bpy.props.StringProperty(description="Segment address of a display list to call")
```
I plan to eventually add a description for every property to make it more user friendly, may as well start now.
Those are displayed in the tooltip when hovering over a property, imo it's a great way to document whatever a property is for

**Review by Dragorn421 — COMMENTED** [review]
I did expect `CollectionProperty` to be some work, but not because of fast64, as I said before it will be fine as is, two custom calls is plenty enough.

**Comment by sauraen** [comment]
I've added the descriptions you requested.

**Review by Dragorn421 — APPROVED** [review]

**Pull request merged**